File: app/src/main/java/com/manimaran/wikiaudio/models/WikiWordsWithoutAudio.java
package com.manimaran.wikiaudio.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class WikiWordsWithoutAudio {

    @SerializedName("continue")
    @Expose
    private Offset offset;
    @SerializedName("query")
    @Expose
    private Query query;

    public Offset getOffset() {
        return offset;
    }

    public void setOffset(Offset offset) {
        this.offset = offset;
    }

    public Query getQuery() {
        return query;
    }

    public void setQuery(Query query) {
        this.query = query;
    }

    public static class Offset {

        @SerializedName("cmcontinue")
        @Expose
        private String nextOffset;

        public String getNextOffset() {
            return nextOffset;
        }

        public void setNextOffset(String nextOffset) {
            this.nextOffset = nextOffset;
        }
    }

    public static class Query {

        @SerializedName("categorymembers")
        @Expose
        private List<WikiTitle> wikiTitleList = new ArrayList<>();

        public List<WikiTitle> getWikiTitleList() {
            return wikiTitleList;
        }

        public void setWikiTitleList(List<WikiTitle> wikiTitleList) {
            this.wikiTitleList = wikiTitleList;
        }
    }

    public static class WikiTitle {

        @SerializedName("pageid")
        @Expose
        private Integer pageId;
        @SerializedName("title")
        @Expose
        private String title;

        public Integer getPageId() {
            return pageId;
        }

        public void setPageId(Integer pageId) {
            this.pageId = pageId;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }
    }

}

File: app/src/main/java/com/manimarank/spell4wiki/data/model/WikiWordsWithoutAudio.kt
package com.manimarank.spell4wiki.data.model

import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * Model class for Wiki words without audio(i.e : list of words don't have audio)
 * @property offset OffsetWords?
 * @property query QueryWords?
 * @constructor
 */
data class WikiWordsWithoutAudio(
    @SerializedName("continue")
    var offset: OffsetWords? = null,
    @SerializedName("query")
    var query: QueryWords? = null
)

/**
 * Model class for Next off set information for pagination
 * @property nextOffset String?
 * @constructor
 */
data class OffsetWords(
    @SerializedName("cmcontinue")
    var nextOffset: String? = null
)

/**
 * Model class for Words list
 * @property wikiTitleList List<WikiTitle>
 * @constructor
 */
data class QueryWords(
    @SerializedName("categorymembers")
    var wikiTitleList: List<WikiTitle> = ArrayList()
)

/**
 * Model class for Words Title in wiktionary
 * @property pageId Int?
 * @property title String?
 * @constructor
 */
data class WikiTitle(
    @SerializedName("pageid")
    var pageId: Int? = null,
    @SerializedName("title")
    var title: String? = null
)